### PR TITLE
Improve docu and switch linter from black to ruff

### PR DESCRIPTION
**Pull request by nquetschlich**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -251,6 +251,8 @@ from mqt.bench import get_benchmark
 qc = get_benchmark("dj", "alg", 5)
 ```
 
+For examples can be found in the `notebooks/examples.ipynb` jupyter notebook.
```
Comment (burgholzer) [inline README.md:254]:
```suggestion
Examples can be found in the `notebooks/examples.ipynb` jupyter notebook.
```
Additionally, you can actually link to the respective file here.

File: notebooks/examples.ipynb
Comment (burgholzer) [inline notebooks/examples.ipynb:1]: Could be worth adding circuit outputs (`qc.draw(output="mpl")` to these cells to make the documentation look a little nicer

File: setup.py
Comment (burgholzer) [inline setup.py:1]: I am fairly sure you can just remove this file entirely by now 🙂

**Review by burgholzer — CHANGES_REQUESTED** [review]
That was quick, many thanks!
Just a couple of small details that should be equally fast to address 🙂

File: README.md (hunk 1 above)
Comment (nquetschlich) [reply README.md]: done

File: notebooks/examples.ipynb
Comment (nquetschlich) [reply notebooks/examples.ipynb:1]: done

File: setup.py
Comment (nquetschlich) [reply setup.py:1]: done

**Review by burgholzer — APPROVED** [review]
At some point it would be nice to move notebooks like the examples one to the RtD documentation (and properly link to it).

**Pull request merged**